src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IProjectFile.cs

```diff
@@ -35,6 +35,8 @@ public interface IProjectFile
 
         void AddItem(string name, string path);
 
+        bool RemoveItem(ProjectItemType? itemType = null, string? includePath = null, string? excludePath = null, string? elementName = null);
```

Comment:
Should the itemname that needs to be removed not be not nullable here? How else can you check what exactly needs to be removed?

Reply:
The parameters act like a filter. This method should remove all items that fit the filter.
If I wanna remove all items with a certain IncludePath, I only specify that. If I want a to remove a particular elementName and an ItemType, with that include path, I specify them too.

Reply:
Perhaps we could consider a `ItemFilter` _type_ so
1. It conveys that the things you set on the object act like an `and` filter set
2. It avoids changing the method on the interface in the future if we add/remove filter capabilities